velero-plugin-for-microsoft-azure/delegationsas.go

```diff
@@ -0,0 +1,220 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+)
+
+type KeyInfo struct {
+	Start  string
+	Expiry string
+}
+
+type UserDelegationKey struct {
+	SignedOid     string
+	SignedTid     string
+	SignedStart   string
+	SignedExpiry  string
+	SignedService string
+	SignedVersion string
+	Value         string
+}
+
+// For more information, see https://docs.microsoft.com/en-gb/rest/api/storageservices/create-user-delegation-sas
+type BlobDelegationSASSignatureValues struct {
```

Comment:
Seems that the upstream has supported the SAS in the latest version https://github.com/Azure/azure-sdk-for-go/tree/sdk/storage/azblob/v0.5.0/sdk/storage/azblob/sas,
could you take a look at it?

Reply:
The code now uses the new version and the custom sas code was removed.